server/lib/device/device.notify.js

```diff
@@ -0,0 +1,44 @@
+const get = require('get-value');
+
+const logger = require('../../utils/logger');
+const { EVENTS } = require('../../utils/constants');
+
+const FUNC_BY_EVENT = {
+  [EVENTS.DEVICE.CREATE]: 'onNewDevice',
+  [EVENTS.DEVICE.UPDATE]: 'onUpdateDevice',
+  [EVENTS.DEVICE.DELETE]: 'onDeleteDevice',
+};
+
+/**
+ * @description Notify a device is created.
+ * @param {Object} device - Created device.
+ * @param {string} event - Event to send.
+ * @example
+ * device.notify({ service_id: 'a810b8db-6d04-4697-bed3-c4b72c996279'}, 'device.create');
+ */
+function notify(device, event) {
+  logger.debug(`Notify device ${device.selector} creation`);
+
+  const serviceFuncName = FUNC_BY_EVENT[event];
+
+  if (!serviceFuncName) {
+    logger.warn(`Event ${event} not handled to notify device changes`);
+  } else {
+    // send global event
+    this.eventManager.emit(event, device);
+
+    // notify concerned service
+    const service = this.serviceManager.getServiceById(device.service_id);
+    if (service === null) {
+      logger.warn(`Service ${device.service_id} was not found.`);
+    } else if (typeof get(service, `device.${serviceFuncName}`) !== 'function') {
+      logger.warn(`Function device.${serviceFuncName} in service ${service.name} does not exist.`);
+    } else {
+      service.device[serviceFuncName](device);
```

Comment:
What happens if the function in the service throw an error ?

It shouldn't make the initial create/update/delete device fail.

Quick question: why did you choose to do a function call, and not an event here? (this is not a trap question :D just to understand your choice!)

Reply:
I'm not sure this should undo the device modification.
According to my vision, this is an non blocking action, as it a "post create" action, handled outside the core "classical" process. But this can change.

Why not using an event on device related service:
1. I think this could be an extension of the service definition for developers (as setValue or poll), so no need to explicitly subscribe to any event if this need is required on the service -> make dev easier
2. to limit event usage (and nodejs MaxListeners warn)
3. as we only want to suscbribe to bluetooth events, the event name will looks like "bluetooth_device_created", so for all services, we'll get a specific event name (eg. "philips-hue_device_created"), this could be a source of errors (typos in event name for instance)

But the global event is still an event, so thrid party services can handle it.

Reply:
Ok, good point.

I'm ok with functions, but as least please add a try catch around the function call !

Reply:
Changes on 2nd commit.